### PR TITLE
feat(web-components): add 'makeAgoricKeplrConnection' util

### PR DESCRIPTION
**Pull request by samsiegart**
refs https://github.com/Agoric/agoric-sdk/issues/6294

Creates a utility function for connecting to keplr and loading the user's smart wallet (just purses for now) without the bridge.

Short demo of it working with a PSM UI with some minimal changes:
keplrpurses

### CODE REVIEW

**Comment by samsiegart** [comment]
I think we should just remove this dapp-treasury build CI step. We already expect the dapp to be broken currently. Is that reasonable @michaelfig?

**Comment by michaelfig** [comment]
> I think we should just remove this dapp-treasury build CI step. We already expect the dapp to be broken currently. Is that reasonable @michaelfig?

Yeah, that's fine for now.

File: packages/web-components/src/keplr-connection/KeplrConnection.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+// @ts-check
+import { makeLeader } from '@agoric/casting';
+import { makeImportContext } from '@agoric/wallet/api/src/marshal-contexts';
+import { getKeplrAddress } from './getKeplrAddress';
+import { getChainId } from './getChainId';
+import { watchWallet } from './watchWallet';
+
+const DEFAULT_NETWORK_CONFIG = 'https://ollinet.agoric.net/network-config';
```
Comment (michaelfig) [inline packages/web-components/src/keplr-connection/KeplrConnection.js:10]: Deserves a TODO:
```suggestion
// TODO: We need a way to detect the appropriate network-config, and default it to mainnet.
const DEFAULT_NETWORK_CONFIG = 'https://ollinet.agoric.net/network-config';
```

File: packages/web-components/src/keplr-connection/errors.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+// @ts-check
+
+export const Errors = {
+  enableKeplr: '0',
+  networkConfig: '1',
+  noSmartWallet: '2',
```
Comment (michaelfig) [inline packages/web-components/src/keplr-connection/errors.js]: Suggest more searchable constants:
```suggestion
  enableKeplr: 'KEPLR_CONNECTION_ERROR_ENABLE_KEPLR',
  networkConfig: 'KEPLR_CONNECTION_ERROR_NETWORK_CONFIG',
  noSmartWallet: 'KEPLR_CONNECTION_ERROR_NO_SMART_WALLET',
```

File: packages/web-components/src/keplr-connection/followLatest.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
+// @ts-check
+import { iterateEach } from '@agoric/casting';
+import { AmountMath } from '@agoric/ertp';
+
+export const followLatest = async ({startingHeight, latestFollower, updatePurses, brandToPurse}) => {
```
Comment (michaelfig) [inline packages/web-components/src/keplr-connection/followLatest.js]:
```suggestion
// TODO: Provide a way for the caller to terminate the follower loop.
export const followLatest = async ({startingHeight, latestFollower, updatePurses, brandToPurse}) => {
```

**Review by michaelfig — APPROVED** [review]
Approach looks good.  Since this is additive, I'm okay approving even if some of the important details are still being worked out.

File: packages/web-components/src/keplr-connection/followLatest.js (hunk 3 above)
Comment (samsiegart) [reply packages/web-components/src/keplr-connection/followLatest.js]: Done. Is there a good way to cancel a follower? Maybe set a flag and break on the next update?

File: packages/web-components/src/keplr-connection/errors.js (hunk 2 above)
Comment (samsiegart) [reply packages/web-components/src/keplr-connection/errors.js]: Done

File: packages/web-components/src/keplr-connection/KeplrConnection.js (hunk 1 above)
Comment (samsiegart) [reply packages/web-components/src/keplr-connection/KeplrConnection.js:10]: Done. What were you thinking by "detect" here? I was thinking the dapp could just be responsible for providing it.

**Pull request merged**